src/Workspaces/Core/Portable/Shared/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -854,5 +854,49 @@ public static bool IsEnumType(this ITypeSymbol type)
         {
             return type.IsValueType && type.TypeKind == TypeKind.Enum;
         }
+
+        public static bool? IsMutableValueType(this ITypeSymbol type)
+        {
+            if (type.IsErrorType())
+            {
+                return null;
+            }
+
+            if (type.TypeKind != TypeKind.Struct)
+            {
+                return false;
+            }
+
+            if (type.IsNullable())
+            {
+                // Nullable<T> can only be mutable if T is mutable. This case ensures types like 'int?' are treated as
+                // immutable.
+                var typeArguments = type.GetTypeArguments();
+                if (typeArguments.Length != 1)
+                {
+                    return null;
+                }
```

Comment:
i'm virtually certain that this can't happen. Indeed, all we're doing is checking for System_Nullable_T.  If you look at the compiler, for example, they assume all over the place that this means you certainly can safely get that type arg back.

The main reason i'm complaining here is that this means we now have this weird null/tri-state pattern for this method.  I'd rather us not have to have that.  Worst case, even if we're paranoid and we see this, we just default to assuming it's mutable (or immutable) or something.  It's basically fine to have this be a boolean and not a tristate.

Reply:
> The main reason i'm complaining here is that this means we now have this weird null/tri-state pattern for this method.

Error type handling leaves us with tri-state even if this goes away :\

Reply:
> i'm virtually certain that this can't happen.

Applied this simplification

> we just default to assuming it's mutable (or immutable) or something

Pattern in similar methods appears to be "when unknown, assume *not* that kind". This causes a problem for the two analyzers which would start to have diagnostic false positives using that approach, so rather than silently using an inconsistent `bool` pattern to make the analyzers work right I changed up the signature to make it obvious that something's different.